### DISCUSSION TITLE
zbm 2.2.3: How to disable USB teardown in kcl?

### DISCUSSION THREAD

**Discussion by gc-user** [zbm-dev/zfsbootmenu#555] [Questions] [answered]
With v2.2.3, USB teardown has been enabled by default (at least on void linux).

How can it be disabled?

Reading the [docs](https://docs.zfsbootmenu.org/en/v2.3.x/man/zfsbootmenu.7.html) I found the kcl entry zbm.skip_hooks=<hooklist>.

Unfortunately, as I don't know the syntax or name for <hooklist>, I had to use try and error, but so far without luck.

I tried adding the following to the end of the kcl:

zbm.skip_hooks=teardown
zbm.skip_hooks=teardown.d
zbm.skip_hooks=90-xhci-unbind.sh

Then I noticed this sentence in the docs: "Only base names of hooks (i.e., with any other path component removed) are matched against the hooklist." and also tried:

zbm.skip_hooks=xhci-unbind
zbm.skip_hooks=unbind

Well, i didn't try zbm.skip_hooks=xhci, though, I just realized...

Also, re-reading this paragraph of the docs, at least the first two attempts, not containing any file names, did fail unsurprisingly.

The only way I was so far successful is by commenting-out the 90-xhci-unbind.sh script and recompiling zbm. But that is not very elegant and won't survive any updates. That's why I'd like to ask for the correct kcl syntax for disabling usb teardown.

Does this kcl entry maybe need to come before the spl.spl_hostid=... entry tin the kcl in order to be executed at the right time?

Thank you for your help!

**Comment by zdykstra** [comment, marked as answer]
You need to set that option on the kernel command line for ZFSBootMenu, not on your boot environment. If you're making an EFI bundle, it can be set in `/etc/zfsbootmenu/config.yaml` - if you're booting a kernel and initramfs via an EFI entry, you'll have to add it there.

The basename of a file just removes any path components. In this case, it would be `zbm.skip_hooks=90-xhci-unbind.sh`.

**Reply by gc-user** [reply]
Wow, thanks for the fast reply, Zach!

My rationale for the BE kcl was "because it happens after the zbm"... 

Thanks for the extended explanation! Now I'm starting to get a few things at the same time!
